kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardHolderLoadServiceImpl.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    /**
+     * Validates and parses the given file, then stores procurement cardholders in a temp table.
+     * 
+     * @param fileName
+     *            The name of the file to be parsed.
+     * @return This method always returns true. An exception is thrown if a problem occurs while loading the file.
+     */
+    @Override
+    public boolean loadProcurementCardHolderFile(String fileName) {
+        FileInputStream fileContents;
+        try {
+            fileContents = new FileInputStream(fileName);
+        } catch (FileNotFoundException e1) {
+            LOG.error("file to parse not found " + fileName, e1);
+            throw new RuntimeException("Cannot find the file requested to be parsed " + fileName + " " + e1.getMessage(), e1);
+        }
+
+        Collection<ProcurementCardHolderLoad> pcardHolders = null;
+        try {
+            byte[] fileByteContent = IOUtils.toByteArray(fileContents);
+            pcardHolders = (Collection<ProcurementCardHolderLoad>) batchInputFileService.parse(procurementCardHolderInputFileType, fileByteContent);
+        } catch (IOException e) {
+            LOG.error("error while getting file bytes:  " + e.getMessage(), e);
+            throw new RuntimeException("Error encountered while attempting to get file bytes: " + e.getMessage(), e);
+        } catch (ParseException e) {
+            LOG.error("Error parsing xml " + e.getMessage());
+            throw new RuntimeException("Error parsing xml " + e.getMessage(), e);
+        }
+
+        if (pcardHolders == null || pcardHolders.isEmpty()) {
+            LOG.warn("No PCard Holders in input file " + fileName);
+        }
+
+        loadTransactions((List<ProcurementCardHolderLoad>) pcardHolders);
+
+        LOG.info("Total holders loaded: " + pcardHolders.size());
+        return true;
+    }
+    
+    @Override
+    public boolean loadFiles() {
+        cleanTransactionsTable();
+
+        List<String> fileNamesToLoad = batchInputFileService.listInputFileNamesWithDoneFile(procurementCardHolderInputFileType);
+
+        //process only most recent file set to prevent duplicates
+        Map<Date, String> fileNameMap = buildMapForFileNameList(fileNamesToLoad);
+        Date holdDate = null;
+        for (Iterator<Date> iter2 = fileNameMap.keySet().iterator(); iter2.hasNext();) {
+            Date fileDate = (Date) iter2.next();
+            if (holdDate == null || holdDate.compareTo(fileDate) < 0) {
+                holdDate = fileDate;
+            }
+        }
+        String holdDateString = (String) fileNameMap.get(holdDate);
+        
+        boolean processSuccess = true;
+
+        for (String inputFileName : fileNamesToLoad) {
+            if (inputFileName.contains(holdDateString)) {
+                // Removing related .done file right away
+                removeDoneFile(inputFileName);
+                LOG.info("Loading Procurement Cardholder file " + new File(inputFileName).getName());
```

Comment:
Please remove the logging statement as the deletion of the file is already logged in removeDoneFile.